### ISSUE TITLE
[Help needed] Page not found "/" route

### ISSUE THREAD

**Issue by BonnieDoug** [nuxt/nuxt.js#2625]
First off, sorry if this is in the wrong place. [I tried StackOverflow first but got no interest. ](https://stackoverflow.com/questions/48301321/page-not-found-route)
I'm trying to set up a VueJS app using Nuxt for server-side rendering. However, after deployment to my server the index.vue is returning a 404 Page not found error.

This doesn't happen when running on my development machine, and still happens even if run in Development mode on my server.

All other routes work, and getting to the index route from within the app itself works fine. It just doesn't load when refreshed. Eg:

`http://myapp.com/` Doesn't work

`http://myapp.com/login` Works fine

My pages folder currently looks like this:

    - pages
    -- index.vue
    -- login.vue

I have nothing fancy set up within my nuxt.config file and have pretty much the same setup as is described in the [Auth0 example][1]

On my server I'm running nuxt with the command `pm2 start npm --name "my-app" -- start` Which runs fine without any errors, and I have the following .htaccess config:

    RewriteCond %{HTTP_HOST} ^my-app\.com$ [OR]
    RewriteRule ^(.*) "http\:\/\/127\.0\.0\.1\:3000\/$1" [P,L]

This is the error screen I get:

[![enter image description here][2]][2]

The page itself is very basic at the moment, it did have far more on it, however, I've cut it down to the following trying to debug this issue:

    <template>
        <v-container grid-list-md>
           <v-layout row wrap>
               <h1>Test index, trying to fix the 404</h1>
               <h2>Hello, {{ loggedUser ? loggedUser : 'friend' }}!</h2>
           </v-layout>
       </v-container>
    </template>

    <script>
     import { mapGetters } from 'vuex'

     export default {
       computed: mapGetters([
         'isAuthenticated',
         'loggedUser'
       ])
     }
    </script>

  [1]: https://github.com/nuxt/example-auth0/blob/master/nuxt.config.js
  [2]: https://i.stack.imgur.com/V4pgP.png

**Comment by Atinux** [comment]
Hi @BonnieDoug 

Does it work if you don't start your server with pm2 but directly with `npm start`?

**Comment by BonnieDoug** [comment]
No, same issue. Tried running it in dev moe (`yarn run dev`) with the same result, was hoping for a more helpful error at least, but get the exact same thing.

**Comment by Cogitoergo** [comment]
@Atinux ,

I could back this up, has nothing to do with PM2.

It happens randomly, sometimes I'm just developing with `npm run dev` and all of a sudden I'm getting like 20 ` { statusCode: 404, message: 'This page could not be found' }` errors, then the open tab freezes. Sometimes I get some errors in console, like "Loading chunk 2 failed" and the Server-side/Client-side Nodes missmatch one.

**Comment by BonnieDoug** [comment]
@Atinux will do. 
@Cogitoergo it's not random, it happens every time on live without fail.

**Comment by Cogitoergo** [comment]
For me its random. Sometimes just saving the file triggers it, sometimes refreshing the page a few times.

Tho I'm always having a console full of `{ statusCode: 404, message: 'This page could not be found' }` on every page. I'm using nuxt-starter-template with vuetify a-la-carte, though only drawer and toolbar imported.

**Comment by LordGuard** [comment]
@Cogitoergo, I have the same error

**Comment by BonnieDoug** [comment]
My fix, isn't a fix. It's started to cause issues with MIMETYPE & EventSource, and also disappearing elements, due to Server Side and Client side html discrepancies.

**Comment by Cogitoergo** [comment]
I finally found the root of the Page Not Found problem. It was coming from missing source maps. I was using Vuetify, but I didin't add it to my vendor array, this cause it to bug out. Either by disabling source maps completely or by adding Vuetify to vendors the problem is now gone.

**Comment by kkojot** [comment]
@Cogitoergo could you tell us how to add vuetify to vendor array? I have exactly the same issue with nuxt+vuetify+a-la-carte and the missing source maps errors. 

In my `nuxt.config.js` I have `vendor: ['~/plugins/vuetify.js', 'vue-i18n'],` and the vuetify plugin is set there by default. What should I do to solve the Page Not Found problem?
